src/middleware/index.ts

```diff
@@ -1,13 +1 @@
-import { UiState } from '../types';
-
-export type MiddlewareDefinition = {
-  onStateChange(options: { uiState: UiState }): void;
-  subscribe(): void;
-  unsubscribe(): void;
-};
-
-export type Middleware = ({
-  instantSearchInstance: InstantSearch,
-}) => MiddlewareDefinition;
-
-export { createRouter, RouterProps } from './createRouter';
```

Comment:
not sure if anyone uses this, but this changes the API

Reply:
I added them back https://github.com/algolia/instantsearch.js/pull/4446/commits/ad0dafbf0e763d589d4da898e7b0431de022ef57
but not like before because of [this](https://github.com/algolia/instantsearch.js/pull/4463/files/c97960e3384f39db2677611b5a5f876e0f66f2a7#diff-f5dcfa3be457d40ab6df1ec1ef46c014)
(tl;dr -> rollup failed to resolve `RouterProps` type from `./createRouter`. It didn't happen before because this was not exported at `src/lib/main.ts`, but not it is.)